### ISSUE TITLE
Dynamic staggerConfig?

### ISSUE THREAD

**Issue by mattrothenberg** [aholachek/react-flip-toolkit#101]
Howdy!

I'm troubleshooting an issue in `vue-flip-toolkit` (https://github.com/mattrothenberg/vue-flip-toolkit/issues/17) where a user has noticed that, due to my hasty implementation, `staggerConfig` is only ever passed to the `Flipper` component during construction and hence doesn't update as a user modifies it.

```vue
// Flipper.vue
mounted() {
  this.flipInstance = new Flipper({
    element: this.$el,
    spring: this.spring,
    ...(this.staggerConfig ? { staggerConfig: this.staggerConfig } : null)
  });
},
```

For example, if a user wants to toggle `reverse` (from true to false, as a function of the "focused" element), the `Flipper` component won't be aware of the change and hence won't stagger the transition. You can see how this impacts the animation in the following Storybook story.

https://keen-davinci-e49eba.netlify.com/?path=/story/examples--accordion

In Vue, there's the concept of "watchers" (similar to `componentDidUpdate`) where you can respond to when a prop's value changes. I tried setting a new `staggerConfig` value on `this.flipInstance` accordingly, but I'm not noticing a change in behavior which makes me wonder whether this setter actually exists/does anything.

```vue
watch: {
  staggerConfig(oldConfig, newConfig) {
    if (oldConfig !== newConfig) {
      console.log("staggerConfig change");
      this.flipInstance.staggerConfig = newConfig;
    }
  }
}
```

Can you recommend a different way of updating my `Flipper` instance's `staggerConfig`?

Hope this makes sense :grin:

**Comment by mattrothenberg** [comment]
Made some progress! After playing around with the markup in my example, I got things to stagger.

I noticed, tho, that the `delayUntil` attribute never gets added to a particular element's `dataset` attribute, and I _think_ that stems from a discrepancy between the React & Vanilla implementations.

[Vanilla](https://github.com/aholachek/react-flip-toolkit/blob/master/packages/react-flip-toolkit/src/FlipToolkit/Flipper.ts#L108-L126)
```js
const flipConfig = {
      scale,
      translate,
      opacity,
      transformOrigin,
      spring,
      stagger
    }
...
element.dataset.flipConfig = JSON.stringify(flipConfig)
// I don't think the delayUntil var is getting added here
```

Curious what you think!

**Comment by aholachek** [comment]
Hi @mattrothenberg,

You're totally right, looks like `delayUntil` is just not getting passed in to `Flipper`. I added the `delayUntil` option a couple of months ago when I refactored to remove nested staggering support, so I must have forgotten to add it to `flip-toolkit`. I'll confirm that and then release a patch.

**Comment by aholachek** [comment]
Just to clarify, do you have stagger updating working or is that something I should also take a look at?

**Comment by mattrothenberg** [comment]
Thanks, and sorry for the confusing ramble.

Yup, I believe I have the dynamic stagger working. The setter function on the Flip instance seems to be doing the trick (at least to my eyes – take a look at the GIF below).

![Kapture 2019-11-09 at 10 48 10](https://user-images.githubusercontent.com/5148596/68531258-91c7b100-02de-11ea-93e7-e9b3cedbdd0e.gif)

Once that `delayUntil` value gets passed to `Flipper`, I'd expect the animation above to change such that the "avatar" doesn't start animating until the parent card expands to its expanded height.

**Comment by aholachek** [comment]
I just published 7.0.7 which should fix the `delayUntil` option, thank you for flagging it, and please let me know if you find any other issues.

**Comment by mattrothenberg** [comment]
Thanks for the quick turnaround @aholachek.
